packages/trpc/server/routers/viewer/apps.tsx

```diff
@@ -0,0 +1,277 @@
+      z.object({
+        slug: z.string(),
+        enabled: z.boolean(),
+      })
+    )
+    .mutation(async ({ ctx, input }) => {
+      console.log("🚀 ~ file: apps.tsx:114 ~ .mutation ~ input", input);
+      const { prisma } = ctx;
+
+      // Get app name from metadata
+      const localApps = getLocalAppMetadata();
+      // console.log("🚀 ~ file: apps.tsx:118 ~ .mutation ~ localApps", localApps);
+      const appMetadata = localApps.find((localApp) => {
+        console.log("🚀 ~ file: apps.tsx:124 ~ appMetadata ~ localApp", localApp.slug);
+
+        return localApp.slug === input.slug;
+      });
+
+      if (!appMetadata)
+        throw new TRPCError({ code: "INTERNAL_SERVER_ERROR", message: "App metadata could not be found" });
+
+      const app = await prisma.app.upsert({
+        where: {
+          slug: input.slug,
+        },
+        update: {
+          enabled: !input.enabled,
+        },
+        create: {
+          slug: input.slug,
+          dirName: appMetadata?.dirName || "",
+          categories:
+            (appMetadata?.categories as AppCategories[]) ||
+            ([appMetadata?.category] as AppCategories[]) ||
+            undefined,
+          keys: undefined,
+        },
+      });
+
+      // If disabling an app then we need to alert users basesd on the app type
+      if (input.enabled) {
+        if (app.categories.some((category) => ["calendar", "video"].includes(category))) {
+          // Find all users with the app credentials
+          const appCredentials = await prisma.credential.findMany({
+            where: {
+              appId: app.slug,
+            },
+            select: {
+              user: {
+                select: {
+                  email: true,
+                  locale: true,
+                },
+              },
+            },
+          });
+
+          Promise.all(
+            appCredentials.map(async (credential) => {
+              const t = await getTranslation(credential.user?.locale || "en", "common");
+
+              if (credential.user?.email) {
+                await sendDisabledAppEmail({
+                  email: credential.user.email,
+                  appName: appMetadata?.name || app.slug,
+                  appType: app.categories,
+                  t,
+                });
+              }
+            })
+          );
+        } else {
+          const eventTypesWithApp = await prisma.eventType.findMany({
+            where: {
+              metadata: {
+                path: ["apps", app.slug as string, "enabled"],
+                equals: true,
+              },
+            },
+            select: {
+              id: true,
+              title: true,
+              users: {
+                select: {
+                  email: true,
+                  locale: true,
+                },
+              },
+              metadata: true,
+            },
+          });
+
+          Promise.all(
+            eventTypesWithApp.map(async (eventType) => {
```

Comment:
Promise all is suppose to act on returned promises but I'm noticing that we're not returning anything from the `.map` Does this work?

Reply:
The `sendDisabledAppEmail` function returns a promise.

Just tested again and an email is sent to multiple accounts with credentials for the disabled app.